src/popup/pages/PermissionManager.vue

```diff
@@ -0,0 +1,323 @@
+        <BtnMain
+          class="btn"
+          variant="secondary"
+          :to="{ name: 'permissions-settings' }"
+        >
+          {{ $t('pages.permissions.cancel') }}
+        </BtnMain>
+        <BtnMain
+          class="btn confirm"
+          :disabled="!permissionChanged"
+          @click="onSavePermission"
+        >
+          {{ $t('pages.permissions.confirm') }}
+        </BtnMain>
+      </div>
+      <BtnMain
+        v-if="editView"
+        class="btn"
+        variant="secondary"
+        @click="removePermissions"
+      >
+        <DeleteIcon />
+        {{ $t('pages.permissions.delete') }}
+      </BtnMain>
+    </div>
+  </div>
+</template>
+
+<script>
+import { pick } from 'lodash-es';
+import { mapGetters } from 'vuex';
+import SwitchButton from '../components/SwitchButton.vue';
+import InputAmount from '../components/InputAmountV2.vue';
+import InputField from '../components/InputField.vue';
+import TokenAmount from '../components/TokenAmount.vue';
+import BtnMain from '../components/buttons/BtnMain.vue';
+import DeleteIcon from '../../icons/delete.svg?vue-component';
+
+export default {
+  components: {
+    SwitchButton,
+    InputField,
+    InputAmount,
+    TokenAmount,
+    BtnMain,
+    DeleteIcon,
+  },
+  data() {
+    return {
+      permission: {
+        host: '',
+        name: '',
+        address: false,
+        messageSign: false,
+        transactionSignLimit: 0,
+        transactionSignLimitLeft: 0,
+        transactionSignFirstAskedOn: null,
+      },
+      permissionChanged: false,
+    };
+  },
+  computed: {
+    ...mapGetters([
+      'formatCurrency',
+      'convertToCurrencyFormatted',
+      'currentCurrencyRate',
+    ]),
+    transactionSignLimitError() {
+      return (
+        Number.isNaN(this.permission.transactionSignLimit)
+        || this.permission.transactionSignLimit < 0
+      ) ? { status: 'error' } : null;
+    },
+    host() {
+      return this.$route.params.host;
+    },
+    editView() {
+      return !!this.$route.meta.edit;
+    },
+    selectedAsset() {
+      return {
+        contractId: 'aeternity',
+        symbol: 'AE',
+        current_price: this.currentCurrencyRate,
+      };
+    },
+  },
+  subscriptions() {
+    return pick(this.$store.state.observables, ['balance']);
+  },
+  mounted() {
+    if (this.editView) {
+      if (!this.$store.state.permissions[this.host]) {
+        this.$router.replace({ name: 'not-found' });
+      } else {
+        this.permission = {
+          ...this.$store.state.permissions[this.host],
+          host: this.host,
+        };
+      }
+    }
+
+    this.$watch('permission', (permission) => {
+      this.permissionChanged = (!!permission.name && !!permission.host);
+    }, { deep: true });
+  },
+  methods: {
+    removePermissions() {
+      this.$store.commit('permissions/removePermission', this.host);
+      this.$router.push({ name: 'permissions-settings' });
+    },
+    onSavePermission() {
```

Comment:
The `removePermissions` is plural. This `onSavePermission` is singular. I suggest to stick with one form.